apps/web/src/services/dashboards/shared/dashboard-widget-input-form/dashboard-widget-options-form/DashboardCostWidgetValueOptionDropdown.vue

```diff
@@ -1,55 +1,203 @@
 <script setup lang="ts">
-import { PIconButton, PSelectDropdown } from '@spaceone/design-system';
+import {
+    reactive, watch,
+} from 'vue';
+
+import { PIconButton, PSelectDropdown, PBadge } from '@spaceone/design-system';
 import type {
     AutocompleteHandler,
     SelectDropdownMenuItem,
 } from '@spaceone/design-system/types/inputs/dropdown/select-dropdown/type';
 
+import { SpaceConnector } from '@cloudforet/core-lib/space-connector';
+
+import ErrorHandler from '@/common/composables/error/errorHandler';
+
+import { useWidgetFormStore } from '@/services/dashboards/shared/dashboard-widget-input-form/widget-form-store';
+
 interface Props {
+    optionKey?: string;
     selected?: SelectDropdownMenuItem[];
-    menuHandlers?: AutocompleteHandler[];
     invalid?: boolean;
 }
 
 const props = withDefaults(defineProps<Props>(), {
+    optionKey: '',
     selected: undefined,
-    menuHandlers: undefined,
     invalid: false,
 });
 
 const emit = defineEmits<{(e: 'update:selected', selected: SelectDropdownMenuItem[]): void;
     (e: 'delete'): void;
 }>();
 
-const handleUpdateSelected = (selected: SelectDropdownMenuItem[]) => {
-    emit('update:selected', selected);
+const widgetFormStore = useWidgetFormStore();
+const widgetFormState = widgetFormStore.$state;
+
+const state = reactive({
+    visibleMenu: false,
+    reloadOnMenuHandlerUpdate: false,
+    selectedKey: {} as SelectDropdownMenuItem,
```

Comment:
Suggestion: The code snippet appears slightly awkward. 
What if `selectedKey` is `undefined` or a `SelectDropdownMenuItem`?
Considering that the type specification for `SelectDropdownMenuItem` allows for the possibility of empty cases, it might be more meaningful to handle this possibility explicitly in the code. 
While it's common sense that it should have at least one property, knowing the type specification for such cases would make this implementation more reasonable. Additionally, using `Object.keys(state.selectedKey).length` for the check may feel like a bit of a workaround.